lib/web_apis/web_apis.es6.js

```diff
@@ -0,0 +1,347 @@
+          of: this.BrowserAPI,
+          daoType: 'MDAO',
+        });
+      },
+    },
+    {
+      name: 'browsers',
+      documentation: `A DAO that contains browser names, versions
+        OS, OS versions and their browser Id.`,
+      factory: function() {
+        return this.EasyDAO.create({
+          name: 'BrowserDAO',
+          of: this.Browser,
+          daoType: 'MDAO',
+        });
+      },
+    },
+    {
+      name: 'interfaces',
+      documentation: `A DAO that contains interface names, API names,
+        and their interfaceId`,
+      factory: function() {
+        return this.EasyDAO.create({
+          name: 'InterfaceDAO',
+          of: this.WebInterface,
+          daoType: 'MDAO',
+        });
+      },
+    },
+    {
+      name: 'mlang',
+      factory: function() {
+        return this.ExpressionsSingleton.create();
+      },
+    },
+  ],
+  methods: [
+    {
+      name: 'importAPI',
+      documentation: 'Import interface/API for a given version of browser.',
+      args: [
+        {
+          name: 'browserName',
+          documentation: `The name of imported browser.`,
+          typeName: 'String',
+        },
+        {
+          name: 'browserVersion',
+          documentation: `The version of imported browser.`,
+          typeName: 'String',
+        },
+        {
+          name: 'OSName',
+          documentation: `The name of operating system.`,
+          typeName: 'String',
+        },
+        {
+          name: 'OSVersion',
+          documentation: `The version of operating system.`,
+          typeName: 'String',
+        },
+        {
+          name: 'apiCatalog',
+          documentation: `The web catalog of this version of
+            browser, tested on the given operating system and version.
+            The json object should be the JSON object returned from
+            com.web.catalog.apiExtractor`,
+          typeName: 'JSON',
+        },
+      ],
+      code: function(browserName, browserVersion,
+        OSName, OSVersion, apiCatalog) {
+        let browser = this.Browser.create({
+          browserName: browserName,
+          browserVersion: browserVersion,
+          OS: OSName,
+          OSVersion: OSVersion,
+        });
+        let browserId = browser.id;
+        this.browsers.put(browser);
+        let interfaceNames = Object.keys(apiCatalog);
+        for (let i = 0; i < interfaceNames.length; i += 1) {
+          let interfaceName = interfaceNames[i];
+          for (let j = 0; j < apiCatalog[interfaceName].length; j += 1) {
+            let APIName = apiCatalog[interfaceName][j];
+            let webInterface = this.WebInterface.create({
+              interfaceName,
+              APIName,
+            });
+            let interfaceId = webInterface.id;
+            this.interfaces.put(webInterface);
+            this.browserAPIs.put(this.BrowserAPI.create({
+              browserId,
+              interfaceId,
+              id: `${browserId}.${interfaceId}`,
```

Comment:
My only concern with GUID or seq-no is that it would be nice to have the property that the id be a function of identifying properties. That way different systems operating over the same data construct the same ids.

@kgrgreer Does that seem feasible, or do you anticipate collision issues with `hashCode()` or similar?